soh/src/code/z_actor.c

```diff
@@ -3141,7 +3142,40 @@ void Actor_FreeOverlay(ActorOverlay* actorOverlay) {
 int gMapLoading = 0;
 
 Actor* Actor_Spawn(ActorContext* actorCtx, PlayState* play, s16 actorId, f32 posX, f32 posY, f32 posZ,
-                   s16 rotX, s16 rotY, s16 rotZ, s16 params) {
+                   s16 rotX, s16 rotY, s16 rotZ, s16 params, s16 canRandomize) {
+    
+    uint8_t tryRandomizeEnemy = CVar_GetS32("gRandomizedEnemies", 0) && gSaveContext.fileNum >= 0 && gSaveContext.fileNum <= 2 && canRandomize;
+
+    if (tryRandomizeEnemy) {
+
+        // Hack to remove enemies that wrongfully spawn because of bypassing object dependency with enemy randomizer on.
+        // This should probably be handled on OTR generation in the future when object dependency is fully removed.
+        // Remove bats and skulltulas from graveyard.
+        // Remove octorok in lost woods.
+        if (((actorId == ACTOR_EN_FIREFLY || (actorId == ACTOR_EN_SW && params == 0)) &&
+             play->sceneNum == SCENE_SPOT02) ||
+            (actorId == ACTOR_EN_OKUTA && play->sceneNum == SCENE_SPOT10)) {
+            return NULL;
+        }
+
+        // Hack to change a pot in Spirit Temple that holds a Deku Shield to not hold anything.
+        // This should probably be handled on OTR generation in the future when object dependency is fully removed.
+        // This Deku Shield doesn't normally spawn in authentic gameplay because of object dependency.
+        if (actorId == ACTOR_OBJ_TSUBO && params == 24597) {
+            params = 24067;
+        }
+
+        RandomizedEnemy newEnemy = GetRandomizedEnemy(play, actorId, posX, posY, posZ, rotX, rotY, rotZ, params);
+        actorId = newEnemy.id;
+        posX = newEnemy.posX;
+        posY = newEnemy.posY;
+        posZ = newEnemy.posZ;
+        rotX = newEnemy.rotX;
+        rotX = newEnemy.rotY;
+        rotX = newEnemy.rotZ;
+        params = newEnemy.params;
```

Comment:
part of me thinks it makes sense to refactor `GetRandomizedEnemy` to take in references, so instead of
```c
RandomizedEnemy GetRandomizedEnemy(PlayState* play, int16_t actorId, f32 posX, f32 posY, f32 posZ, int16_t rotX, int16_t rotY, int16_t rotZ, int16_t params);
```
you could use
```c
bool GetRandomizedEnemy(PlayState* play, int16_t& actorId, f32& posX, f32& posY, f32& posZ, int16_t& rotX, int16_t& rotY, int16_t& rotZ, int16_t& params);
```
and then change those values in place, and instead of
```c
RandomizedEnemy newEnemy = GetRandomizedEnemy(play, actorId, posX, posY, posZ, rotX, rotY, rotZ, params);
```
you could have
```c
if (!GetRandomizedEnemy(play, actorId, posX, posY, posZ, rotX, rotY, rotZ, params)) return NULL;
```
in that case the `return NULL` from earlier in this block could be turned into a `return false` within `GetRandomizedEnemy`

i'm not sure if that makes this cleaner or not, but it would cut down on this big block of variable assignments and remove the need to have the `HACK`s within the main `Actor_Spawn` method

Reply:
This function is used in C code so I can't use references. I've tried using pointers but I'm super new to how they work so I haven't gotten far. Will try some more, but if anyone can help me with this it'd be greatly appreciated.

Reply:
OK, I got it to work. I'd love for someone to look it over if I'm doing this the proper way. In-game everything is confirmed working though.